voorhoede/tudelft-spacefinder
# Feat/import aggregated rooms, fix space import issues

# Changes

- All processing of data incoming from the CSV file, the CMS and Exchange is now done without the Ramda library and has been rewritten in TypeScript
- Some minor issues with processing that data have been fixed (e.g. trimming), that previously lead to incorrect calculation of total amount of spaces, seats and generation of building slugs
- A new set of data is now being generated in addition to buildings and spaces - the rooms. A room is identified by a unique real estate number and may contain several spaces. It is assumed that spaces within the same room have the same room number, exchange IDs, quietness and study type settings. The boolean flags for facilities within a room are a result of combining the flags for spaces within the room using OR.
- There is now an .env setting to control whether we want to deal with rooms instead of spaces in the UI (SPACES_MODE=rooms). All relevant components can now work to display either a room or a space. No new patterns appear in the URLs, with /buildings/:buildingSlug/spaces/:spaceSlug leading to a room if a setting says so. For rooms, their real estate number is currently used as slug
- Occupancy data per room is now loaded and subscribed to, but not yet actually used



# Associated issue



# How to test
- https://deploy-preview-157--spacefinder.netlify.app/nl/gebouwen/lr/ redirects to .../62-lr/ and the new URL works (the old slug was generated incorrectly) 
- https://deploy-preview-157--spacefinder.netlify.app/nl/gebouwen/21-lib/ shows >1000 seats and 48 spaces, not 152 seats and 15 spaces like before
- test it with SPACE_MODE=rooms and see how the space URLs are different, total number of spaces is less, coordinates are averaged, total number of rooms, not spaces, per building is displayed.

# Checklist
